Assets/Scripts/PuzzleScript/RotationTarget.cs

```diff
@@ -0,0 +1,14 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class RotationTarget : MonoBehaviour
+{
+    //Is the reference of the LampTarget
+    public Transform target;
+
+    void Update()
```

Comment:
Consider avoiding the use of **Update** methods whenever possible, as they can impact performance. Exploring event-driven or coroutine-based alternatives could help optimize the code further.